File: 22-秋招/笔试/220831-度小满/1.js
function solution(n = 4, k = 2, nums = [4, 3, 2, 7]) {
  let max = nums[0],
    min = nums[0],
    count = 0;

  for (let i = 0; i < n; ++i) {
    for (let j = i + 1; j < n; ++j) {
      min = Math.min(...nums.slice(i, j + 1));
      max = Math.max(...nums.slice(i, j + 1));
      if (max === k * min) {
        count++;
      }
    }
  }

  // let count = 0;

  // for (let i = 0; i < n; i++) {
  //   for (let j = i + 1; j <= n; j++) {
  //     let arr = nums.slice(i, j);
  //     arr.sort((a, b) => a - b);
  //     if (k * arr[0] == arr[arr.length - 1]) {
  //       count++;
  //     }
  //   }
  // }

  return count;
}
const readline = read_line;
// 从一行中读取一列数组
const readArray = () => {
  const line = readline();
  return line.split(" ").map(Number);
};

// 读取一行为一个数字
const readInt = () => {
  return parseInt(readline());
};

function main() {
  const [n, k] = readArray();
  const arr = readArray();

  const res = solution(n, k, arr);
  console.log(res);
}

main();
console.log(solution());

File: 22-秋招/笔试/220831-度小满/1.java
import java.util.*;
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int k = sc.nextInt();
    int[] arr = new int[n];
    for (int i = 0; i < n; i++) {
      arr[i] = sc.nextInt();
    }

    solution(n, k, arr);
  }

  static void solution(int n, int k, int[] nums) {
    int count = 0;

    for (int i = 0; i < n; i++) {
      for (int j = i + 1; j <= n; j++) {
        int[] arr = Arrays.copyOfRange(nums, i, j);
        Arrays.sort(arr);
        if (k * arr[0] == arr[arr.length - 1]) {
          count++;
        }
      }
    }

    System.out.println(count);
  }
}